src/xref2/errors.ml

```diff
@@ -13,12 +13,13 @@ module Tools_error = struct
     | `UnresolvedForwardPath
       (** The module signature depends upon a forward path *)
     | `UnresolvedPath of
-      [ `Module of Cpath.module_ | `ModuleType of Cpath.module_type ]
+      [ `Module of Cpath.module_ * simple_module_lookup_error
+      | `ModuleType of Cpath.module_type * simple_module_type_lookup_error ]
```

Comment:
Allowing these will makes traversing errors more complicated than just looking at `` `Parent ``.

Reply:
The problem is that they are always as a result of another error, which we were always throwing away - in particular here:

https://github.com/ocaml/odoc/blob/8b0a6323d80345f1c60332375417b40886395454/src/xref2/expand_tools.ml#L86

If we do that we risk forgetting that a problem was down to `UnexpandedTypeOf` somewhere. In that case above, the error is a `simple_module_lookup_error` which was `` `Parent_sig (`UnexpandedTypeOf) `` - so we'd have to check everywhere we wanted to return a `` `Module `` or `` `ModuleType `` and return an `` `UnexpandedTypeOf ``. It seemed a bit more error-prone than just making sure we never throw away any Error info.

Reply:
It's fine to go with this. We'll know better next time we need to traverse errors again :)